src/zenml/zen_server/pipeline_deployment/utils.py

```diff
@@ -125,52 +124,62 @@ def run_pipeline(
         deployment_id=new_deployment.id
     )
 
-    def _task() -> None:
-        pypi_requirements, apt_packages = get_requirements_for_stack(
-            stack=stack
-        )
+    placeholder_run = create_placeholder_run(deployment=new_deployment)
+    assert placeholder_run
```

Comment:
Not sure if it's just me, but assertions in productive code rubs me the wrong way. Are we that certain that this will exist? Can't we instead do an if check and throw an error instead? or do a try except around the create_placeholder_run?

Reply:
even if you just end up reraising the same error, which I guess you would wanna do here

Reply:
Yes we're sure, the deployment does not contain a schedule which is the only reason no placeholder run would be created.

I'm not sure what you mean by reraising, there is no error being raised in these lines of code?

Reply:
sorry, I get what you mean, I made the assumption that we need the assertion there because we assume create_placeholder_run() would fail and raise an error in all other cases. I get now that you are making this assertion because create_placeholder_run returns Optional[ResponseModel]. 

I also noticed that this was preexisting code. Won't hold this up anymore.